src/services/nostr/handler.ts

```diff
@@ -173,17 +184,29 @@ export default class Handler {
         this.relay?.close()
         this.relay = null
         this.sub = null
+        this.providerSub?.close()
+        this.providerRelay?.close()
+        this.providerRelay = null
+        this.providerSub = null
     }
 
     async ConnectPromise() {
         return new Promise<void>(async (res) => {
-            this.relay = await this.GetRelay()
+            // Connect to main relay for apps
+            this.relay = await this.GetRelay(this.settings.relays[0], "main")
             if (!this.relay) {
                 res()
                 return
             }
-            this.sub = this.Subscribe(this.relay)
-            this.relay.onclose = (() => {
+            this.sub = this.Subscribe(this.relay, 'app')
+
+            // Connect to provider relay for clients
+            this.providerRelay = await this.GetRelay(this.settings.providerRelay, "provider")
+            if (this.providerRelay) {
+                this.providerSub = this.Subscribe(this.providerRelay, 'client')
+            }
```

Comment:
In here we're just moving forward if the provider relay fails to connect, we might need some retry logic here